include/eve/detail/function/interleave.hpp

```diff
@@ -7,12 +7,30 @@
 //==================================================================================================
 #pragma once
 
+
 #include <eve/arch.hpp>
-#include <eve/detail/overload.hpp>
+#include <eve/traits/overload.hpp>
 
 namespace eve
 {
-  EVE_MAKE_CALLABLE(interleave_, interleave);
+  template<typename Options>
+  struct interleave_t : callable<interleave_t, Options>
+  {
+
+    template<eve::simd_value T, std::same_as<T>... Ts>
+    EVE_FORCEINLINE kumi::tuple<T, Ts...> operator()(T v, Ts... vs) const noexcept
+    { return EVE_DISPATCH_CALL(v, vs...); }
+
+
+    template<simd_value T, std::same_as<T>... Ts>
+    EVE_FORCEINLINE kumi::tuple<logical<T>, logical<Ts>...> operator()(logical<T> v, logical<Ts>... vs) const noexcept
+    requires( T::abi_type::is_wide_logical )
```

Comment:
this should never happen

Reply:
the requries is an implemantiond etail, the () must work for all kind of logicals

Reply:
line suppressed